services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -0,0 +1,244 @@
+import Data.Aeson (encode)
+import Galley.API.Error
+import Brig.Types.Provider
+import Brig.Types.Team.LegalHold
+import Brig.Types.Client.Prekey
+import Control.Monad.Catch
+import Control.Lens (view, (^.))
+import Data.Id
+import Data.ByteString.Conversion (toByteString')
+import Data.Misc
+import Data.LegalHold (UserLegalHoldStatus(..))
+import Galley.API.Util
+import Galley.App
+import Galley.Types.Teams as Team
+import Network.HTTP.Types.Status (status201, status204)
+import Network.Wai
+import Network.Wai.Predicate hiding (setStatus, result, or)
+import Network.Wai.Utilities as Wai
+import UnliftIO.Async (pooledMapConcurrentlyN_)
+
+import qualified Galley.Data                  as Data
+import qualified Galley.Data.LegalHold        as LegalHoldData
+import qualified Galley.Intra.Client as Client
+import qualified Galley.External.LegalHoldService as LHService
+import qualified System.Logger.Class as Log
+
+assertLegalHoldEnabled :: TeamId -> Galley ()
+assertLegalHoldEnabled tid = unlessM (isLegalHoldEnabled tid) $ throwM legalHoldNotEnabled
+
+isLegalHoldEnabled :: TeamId -> Galley Bool
+isLegalHoldEnabled tid = do
+    lhConfig <- LegalHoldData.getLegalHoldTeamConfig tid
+    return $ case legalHoldTeamConfigStatus <$> lhConfig of
+        Just LegalHoldEnabled  -> True
+        Just LegalHoldDisabled -> False
+        Nothing                -> False
+
+-- | Get legal hold status for a team.
+getEnabled :: TeamId ::: JSON -> Galley Response
+getEnabled (tid ::: _) = do
+    legalHoldTeamConfig <- LegalHoldData.getLegalHoldTeamConfig tid
+    pure . json . fromMaybe defConfig $ legalHoldTeamConfig
+  where
+    defConfig = LegalHoldTeamConfig LegalHoldDisabled
+
+-- | Enable or disable legal hold for a team.
+setEnabled :: TeamId ::: JsonRequest LegalHoldTeamConfig ::: JSON -> Galley Response
+setEnabled (tid ::: req ::: _) = do
+    legalHoldTeamConfig <- fromJsonBody req
+    case legalHoldTeamConfigStatus legalHoldTeamConfig of
+        LegalHoldDisabled -> removeSettings' tid Nothing
+        LegalHoldEnabled -> pure ()
+    LegalHoldData.setLegalHoldTeamConfig tid legalHoldTeamConfig
+    pure noContent
+
+createSettings :: UserId ::: TeamId ::: JsonRequest NewLegalHoldService ::: JSON -> Galley Response
+createSettings (zusr ::: tid ::: req ::: _) = do
+    assertLegalHoldEnabled tid
+
+    membs <- Data.teamMembers tid
+    void $ permissionCheck zusr ChangeLegalHoldTeamSettings membs
+
+    newService :: NewLegalHoldService
+        <- fromJsonBody req
+
+    (key :: ServiceKey, fpr :: Fingerprint Rsa)
+        <- LHService.validateServiceKey (newLegalHoldServiceKey newService)
+               >>= maybe (throwM legalHoldServiceInvalidKey) pure
+    LHService.checkLegalHoldServiceStatus fpr (newLegalHoldServiceUrl newService)
+
+    let service = legalHoldService tid fpr newService key
+    LegalHoldData.createSettings service
+    -- TODO: Refactor and use `json` function instead which ensures correct content-type is set, etc.
```

Comment:
Looks easy enough to fix now

Reply:
9b289db9497337811a8a86316c05907b9e4c2163

(it was easy, only took me two commits...  :-))